pylint/checkers/misc.py

```diff
@@ -151,7 +152,7 @@ def process_module(self, module):
         """
         if self.config.notes:
             notes = re.compile(
-                r'.*?#\s*(%s)(:*\s*.*)' % "|".join(self.config.notes), re.I)
+                r'#\s*(%s)\b' % "|".join(map(re.escape, self.config.notes)), re.I)
```

Comment:
Are you sure the word boundary is needed? The new expression is not able to catch anything and right now it passes because we also do an `in` check earlier at line 106, before actually searching with the expression.

Reply:
The word boundary is needed, so that a warning doesn't get emitted about e.g. `# Todoist API`. The commits contain test cases which should cause failures if the changes don't work, and they did fail when I was making these changes, so at least the tests worked back then. But now they pass (also in Travis), so I'm not sure what to say :)

Reply:
Here's a couple of manual tests:
```python
>>> import re
>>> myre = re.compile(r'#\s*(%s)\b' % "|".join(map(re.escape, ['todo', 'fixme'])), re.I)
>>> myre.search("# TODO foo")
<re.Match object; span=(0, 6), match='# TODO'>
>>> myre.search("# Todoist")
>>>
```
At least these work as intended: First matches, second doesn't.

Reply:
@scop I probably forgot to make the string a raw string. Also the test is passing because we do an `in` early on before applying the regular expression (https://github.com/PyCQA/pylint/pull/2320/files#diff-a4b1cfd0674823eef01c61047e6e6bdaL106). I think we should remove that and allow just the regex instead as the performance optimisation feels premature in this case.